galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java

```diff
@@ -67,7 +68,16 @@ public class RunQuery extends HttpServlet {
 
 	@Override
 	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+		try{
+			retrieveResults(req, resp);
+		}catch (Exception e){
+			String msg = new ServletError(GAL5000_GENERIC_API_ERROR).toString();
+					sendResponse(resp, msg, HttpServletResponse.SC_NOT_FOUND);
```

Comment:
Is a 404 the best response when handling a generic error like this?

Reply:
Fixed today with @AaronJhaj